### PR TITLE
#2288: Add new script to generate/fix header guards and license

### LINKED ISSUE #2288: Header guards generation
We have the CI check for header guards correctness (https://github.com/DARMA-tasking/vt/blob/develop/.github/workflows/check-include-guards.yml) 

Let's see if we can reuse the `guardonce` python package for generating the correct header guards. 
This would be used for both `magistrate` and `vt`

### PR DESCRIPTION
**Pull request by JacobDomagala**
Fixes #2288

### CODE REVIEW

**Comment by JacobDomagala** [comment]
@thearusable Can you test the script on the `magistrate` (https://github.com/DARMA-tasking/magistrate/issues/260)?

**Comment by thearusable** [comment]
@JacobDomagala Script looks good!

**Review by thearusable — APPROVED** [review]

File: scripts/generate_header_guards_and_license.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+def generate_license(file_path, license_path):
+    with open(file_path, 'r') as file:
+        lines = file.readlines()
+
+    with open(license_path, 'r') as license_file:
+        license_file = license_file.readlines()
+
+    file_name = os.path.basename(file_path)
+    lenright = int((80 - 2 - len(file_name))/2)
+    license_file[3] = f"//{' ' :<{lenright}}{file_name}\n"
```
Comment (JacobDomagala) [inline scripts/generate_header_guards_and_license.py:53]: - formula for "centering" file name is taken from perl script
- `license_file[3]` can we assume <file_name> placeholder will always be at 3rd line in license template?

**Comment by lifflander** [comment]
We want to change `./scripts/check_guards.sh` to target the new script.

**Comment by JacobDomagala** [comment]
It seems that we will have to remove `check-license` from required workflows.

**Review by lifflander — APPROVED** [review]
Looks good

**Pull request merged**